File: check-if-all-1s-are-at-least-length-k-places-away.rs
pub fn k_length_apart(nums: Vec<i32>, k: i32) -> bool {
    let mut start: i32 = -1;
    let mut end: i32 = -1;

    for i in 0..nums.len() {

        let num = nums[i];

        if num == 1 {
            end = start;
            start = i as i32;

            if end != -1 && start - end - 1 < k {
                return false;
            }
        }

    }
    return true;
}

File: check-if-all-1s-are-at-least-length-k-places-away.ts
{
    function kLengthApart(nums: number[], k: number): boolean {
        let lastOneIndex = -1;
        for (let i = 0 ; i < nums.length; i++) {
            if (nums[i] === 1) {
                if (lastOneIndex >= 0 && i - lastOneIndex - 1 < k) return false;
                lastOneIndex = i;
            }
        }
        return true;
    };   
}